Fix Issue #49

SuprexDE opened the pull request:
Fix https://github.com/LoneDev6/BungeePackFix/issues/49

The problem was that axiom was quite old (April 2022) along with snakeyaml. In the future, I recommend that axiom no longer be used because it is no longer being developed. With this change my players were no longer kicked.

nhanledev commented:
There's a bug by mojang in 1.20.2 that makes the resource pack unload when you switch servers. The patch [[52973c0]](https://github.com/LoneDev6/BungeePackFix/pull/50/commits/52973c07428188318d18d3d12d1c2c15366f472c) doesn't fix it, and it cause the client from loading the resource pack again. I think the plugin should check if the version is 1.20.2 and force reloading the pack (in other words, stop the plugin itself) for that one version only.

SuprexDE commented:
Yes, I know that 1.20.2 doesn't work. That's why my commit says "Currently not working". The MR should fix another error.

SuprexDE commented:
"Where did you get the 1.20.2 packet ID for ClientboundResourcePackPacket? Why did you remove the packet mapping ID for ServerboundResourcePackPacket?"

I have reverted my change to 1.20.2 because otherwise 1.20.2 would no longer load at all.
-> https://wiki.vg/Protocol#Resource_Pack_.28play.29 & https://wiki.vg/Protocol#Resource_Pack_.28configuration.29

I updated the axiom Dependency because the snakeyml dependency within it was too old (as of May 24, 2022). I believe BungeeCord has now upgraded its snakeyaml  to version 2.2, which caused some issues here. The 1.1.3-SNAPSHOT version is up to date (as of April 2023) and is already using snake-yaml version 2.0. However, it is no longer being actively developed.



@LoneDev6

nhanledev commented:
> "Where did you get the 1.20.2 packet ID for ClientboundResourcePackPacket? Why did you remove the packet mapping ID for ServerboundResourcePackPacket?"
> 
> I have reverted my change to 1.20.2 because otherwise 1.20.2 would no longer load at all. -> https://wiki.vg/Protocol#Resource_Pack_.28play.29 & https://wiki.vg/Protocol#Resource_Pack_.28configuration.29
> 
> I updated the axiom Dependency because the snakeyml (version 1.x) dependency within it was too old (as of May 24, 2022). I believe BungeeCord has now upgraded its snakeyaml to version 2.2, which caused some issues here. The 1.1.3-SNAPSHOT version is up to date (as of April 2023) and is already using snake-yaml version 2.0. However, it is no longer being actively developed.
> 
> @LoneDev6

Hello,

Your packet fix is correct and you dont have to revert it. The problem is in the 1.20.2 client and I think gou need to force resend the resource pack from the proxy side instead by detecting the protocol version.

LoneDev6 commented:
Please make a pull request with only the needed edits, thanks
